chapters/chapter-12/application/src/main/java/dev/stratospheric/collaboration/TodoCollaborationService.java

```diff
@@ -69,7 +70,7 @@ public String shareWithCollaborator(String email, Long todoId, Long collaborator
 
     todoCollaborationRequestRepository.save(collaboration);
 
-    queueMessagingTemplate.convertAndSend(todoSharingQueueName, new TodoCollaborationNotification(collaboration));
+    sqsTemplate.sendAsync(todoSharingQueueName, new TodoCollaborationNotification(collaboration));
```

Comment:
Is there a reason to make it async now? A possible con I see here is that we don't get notified if the delivery of the message failed (e.g. missing credentials, missing roles, etc.) as it's now a fire-and-forget.

We can make it async and then maybe add a callback hook for the error case to log the failure?

Reply:
To keep things simple, I'll revert to synchronous transport.